### PR TITLE
fix: schema parsing for v3 

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR fixes a few things:

- Fixes schemas are not parsed when having different schema formats
- Fixes v3 message trait with headers was using incorrect types as they can now be multiformat schemas.

**Related issue(s)**
Fixes https://github.com/asyncapi/parser-js/issues/859

### CODE REVIEW

File: src/custom-operations/parse-schema.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ const customSchemasPathsV2 = [
   '$.components.messages.*',
 ];
 
+const customSchemasPathsV3 = [
+  // operations
+  '$.channels.*.messages.*.payload',
+  '$.components.channels.*.messages.*.payload',
+  '$.components.messages.*.payload',
+  '$.components.schemas.*',
+];
```
Comment (smoya) [inline src/custom-operations/parse-schema.ts:31]: Just wondering if we should not include headers and examples.

```suggestion
const customSchemasPathsV3 = [
  // channels
  '$.channels.*.messages.*.payload',
  '$.channels.*.messages.*.headers',
  '$.components.channels.*.messages.*.payload',
  '$.components.channels.*.messages.*.headers',
  // operations
  '$.operations.*.messages.*.payload',
  '$.operations.*.messages.*.headers',
  '$.components.operations.*.messages.*.payload',
  '$.components.operations.*.messages.*.headers',
  // messages
  '$.components.messages.*.payload',
  '$.components.messages.*.headers.*',
  '$.components.messages.*.examples.*.headers',
  '$.components.messages.*.examples.*.payload',
  // schemas 
  '$.components.schemas.*',
];
```

File: src/models/v3/schema.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export class Schema extends BaseModel<v3.MultiFormatSchemaObject, { id?: string,
 
   $schema(): string {
     if (typeof this._schemaObject === 'boolean') return 'http://json-schema.org/draft-07/schema#';
-    return this._schemaObject.$schema || 'http://json-schema.org/draft-07/schema#';
+    return this._schemaObject.$schema ?? 'http://json-schema.org/draft-07/schema#';
```
Comment (smoya) [inline src/models/v3/schema.ts:50]: Is there an explicit case where you define the `$schema` field but with null or empty string? Just out of curiosity.

File: src/custom-operations/parse-schema.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/custom-operations/parse-schema.ts:31]: But does any of these need custom schema parsing? Headers, you right 👍 Examples, those are actual values not schemas 🤔 

Also `'$.operations.*.messages.*.payload',` is not even a thing right? Unless references are resolved the schema parsing?

File: src/models/v3/schema.ts (hunk 2 above)
Comment (jonaslagoni) [reply src/models/v3/schema.ts:50]: No, at least not to my knowledge 🙂

Comment (smoya) [reply src/models/v3/schema.ts:50]: So then why changing this?

Comment (jonaslagoni) [reply src/models/v3/schema.ts:50]: Because sonar likes it better 🤷

File: src/custom-operations/parse-schema.ts (hunk 1 above)
Comment (smoya) [reply src/custom-operations/parse-schema.ts:31]: Resolution happens **after** applying custom operations (inclulding schema parsing), right?

Comment (jonaslagoni) [reply src/custom-operations/parse-schema.ts]: It did not 😄 

Adapted the paths and tests accordingly.

**Review by smoya — APPROVED** [review]
LGTM!🙌

**Pull request merged**